src/main/java/org/dmfs/oauth2/client/tokens/JsonAccessToken.java

```diff
@@ -125,6 +126,12 @@ public OAuth2Scope scope() throws ProtocolException
         return new Mapped<>(new OAuth2ScopeFunction(), new NullSafe<>(mTokenResponse.optString("scope", null))).value(mScope);
     }
 
+    @Override
+    public Optional<CharSequence> extraParameter(final String parameterName) throws ProtocolException
+    {
+        return new NullSafe<CharSequence>(mTokenResponse.optString(parameterName, null));
```

Comment:
I was worried that `optString` would return `null` if the value is an object, number or boolean. However looking at the code it seems it just calls `toString()` internally, so this is fine.

Reply:
My thoughts exactly. That's why I skipped on using `Mapped` and just put `NullSafe`.